wicket-extensions/src/main/java/org/apache/wicket/extensions/ajax/markup/html/modal/res/modal.js

```diff
@@ -1143,23 +1085,9 @@
 
 				e.style.zIndex = Wicket.Window.Mask.zIndex;
```

Comment:
I believe that dodgy HTML in getMarkup at LL981-1041 can also be heavily simplified. I think it was done that complex to make it drop shadow made from images, which is unnecessary in 2019, because support for browsers without `box-shadow` has dropped a while ago. I can do this refactoring later to make it look nicer as well.

Reply:
modal.js is a beast on its own. IMHO we shouldn't put any effort in it, but deprecate the whole thing instead.

Reply:
You're not going to deprecate the org.apache.wicket.extensions.ajax.markup.html.modal.ModalWindow itself? I think it's good to refactor it with some new JS or maybe to prepare some replacement with similar interface.

Reply:
Lately there are no many new tickets for ModalWindow but there are several old ones in JIRA which no one cared to work on, me included.
I'm +1 to deprecate it but keep it around. The deprecation is just to tell the users "We don't want to maintain it anymore". But I think we should keep it for the ones which use it without problems. Migrating to another modal (Bootstrap or anything) is an adventure on its own.

Reply:
ModalWindow is broken beyond repair - not only its JS, but the Wicket internals are broken too, e.g. the need to wrap a Form around it.

Reply:
+1 we'll keep it around until someone has better ideas.

Reply:
We have written a replacement for wicket modal that is ADA compliant and more flexible. I can rip out our dependencies and commit it. It does use jquery...i guess that part will have to be rewritten if we are migrating away.

Reply:
That's a common issue of wicket. I love wicket, it's brilliant, but most of wicket's JS/HTML/CSS looks like 2005, and it has a bad impact on framework decisions people make, because the first thing people check is demos and resulting HTML, unfortunately, looks like a complete rubbish. I really want to do my best to improve this.

Reply:
> We have written a replacement for wicket modal that is ADA compliant and more flexible. I can rip out our dependencies and commit it. It does use jquery...i guess that part will have to be rewritten if we are migrating away.

That'd be great if the Java API is still the same, or at least similar to the current one.
If it is a totally new implementation then I suggest to discuss it at dev@.

Reply:
> That's a common issue of wicket. I love wicket, it's brilliant, but most of wicket's JS/HTML/CSS looks like 2005

Are you talking about the HTML generated by Wicket(-core/-extensions) components ? Or about wicket-examples ?

Because Wicket generates whatever the application tells it to generate.
Wicket-examples received some styling updates for 8.0.0 but any help to improve it further is very welcome!

Reply:
I mean wicket-ajax-jquery particularly. It's a 3000 lines file in the source, it has three window.eval, the wicket modal, mentioned above. Demos are also not great.

Reply:
> I mean wicket-ajax-jquery particularly. It's a 3000 lines file in the source, it has three window.eval,

Managers do not care about this, do they ? I think only geeks care about this.

Reply:
Well, true. But it won't even be included in the list by technical team. At least that's what happened to myself in the past, I considered wicket but then was scared away by demos and wicket-ajax-jquery, so I didn't include it into the list. Also necessity to include 'unsafe-eval' into csp might drop chances of approval to zero if there's a security team present.

Reply:
> ... I considered wicket but then was scared away by demos

@andruhon I guess your are referring to our [examples site](http://examples8x.wicket.apache.org/index.html). These are mainly technical examples for teaching purpose and yes, they are not the best good-looking examples of Wicket applications. 
There are however a couple of community sites that host more advanced examples using popular web frameworks like [Bootstrap](https://wb7.teliclab.info), [JQuery UI and Kendo UI](http://www.7thweb.net/wicket-jquery-ui). 
Maybe these two should be more visible on our official site (http://wicket.apache.org/).

Reply:
All I wanted to tell is that wicket core's JS needs some attention, wicket-ajax in particulars, and I'm volunteering to discuss and contribute.